lib/rules/amd-function-arity.js

```diff
@@ -54,6 +83,16 @@ module.exports = function (context) {
                 expected: dependencyCount,
                 actual: actualParamCount
             });
+        } else if (dependencyNodes.length > callbackParams.length && allowedExtraDependencies && !allowExtraDependencies) {
```

Comment:
With the single `allowExtraDependencies` variable approach mentioned above, your test here would become:
```js
dependencyNodes.length > callbackParams.length && Array.isArray(allowExtraDependencies)
```